x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/analytics/action/PostAnalyticsEventAction.java

```diff
@@ -0,0 +1,248 @@
+            this.payload = in.readBytesReference();
+        }
+
+        public String eventCollectionName() {
+            return eventCollectionName;
+        }
+
+        public AnalyticsEvent.Type eventType() {
+            return AnalyticsEvent.Type.valueOf(eventType.toUpperCase(Locale.ROOT));
+        }
+
+        public long eventTime() {
+            return eventTime;
+        }
+
+        public BytesReference payload() {
+            return payload;
+        }
+
+        public boolean isDebug() {
+            return debug;
+        }
+
+        public XContentType xContentType() {
+            return xContentType;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            ActionRequestValidationException validationException = null;
+
+            if (Strings.isNullOrEmpty(eventCollectionName)) {
+                validationException = addValidationError("collection name is missing", validationException);
+            }
+
+            if (Strings.isNullOrEmpty(eventType)) {
+                validationException = addValidationError("event type is missing", validationException);
+            }
+
+            try {
+                if (eventType.toLowerCase(Locale.ROOT).equals(eventType) == false) throw new IllegalArgumentException();
+
+                AnalyticsEvent.Type.valueOf(eventType.toUpperCase(Locale.ROOT));
+            } catch (IllegalArgumentException e) {
+                validationException = addValidationError(
+                    LoggerMessageFormat.format("invalid event type: {}", eventType),
+                    validationException
+                );
+            }
+
+            return validationException;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeString(eventCollectionName);
+            out.writeString(eventType);
+            out.writeBoolean(debug);
+            out.writeLong(eventTime);
+            XContentHelper.writeTo(out, xContentType);
+            out.writeBytesReference(payload);
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Request that = (Request) o;
+            return Objects.equals(eventCollectionName, that.eventCollectionName)
+                && debug == that.debug
+                && eventTime == that.eventTime
+                && Objects.equals(eventType, that.eventType)
+                && Objects.equals(xContentType, that.xContentType)
+                && Objects.equals(payload, that.payload);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(eventCollectionName, eventType, debug, eventTime, xContentType, payload);
+        }
+    }
+
+    public static class Response extends ActionResponse implements ToXContentObject {
```

Comment:
You could consider breaking this out into `DefaultResponse` and `DebugResponse`.

Reply:
I like that idea. DebugResponse could be a subclass of DefaultResponse.

Reply:
The class has been split in two different classes.
I did not used an `AcknowledgedResponse` because it is a response type that is meant to be used in the context of   master requests.